File: python/08_033.py
def max_viewing_area(heights):

    left, right = 0, len(heights) - 1
    max_area = 0
    
    while left < right:
        width = right - left
        height = min(heights[left], heights[right])
        max_area = max(max_area, width * height)
        
        if heights[left] < heights[right]:
            left += 1
        else:
            right -= 1
            
    return max_area

File: cpp/08_033.cpp
#include <vector>
#include <algorithm>

int maxViewingArea(std::vector<int>& heights) {
    int left = 0;
    int right = heights.size() - 1;
    int maxArea = 0;
    
    while (left < right) {
        int width = right - left;
        int height = std::min(heights[left], heights[right]);
        maxArea = std::max(maxArea, width * height);
        
        if (heights[left] < heights[right]) {
            left++;
        } else {
            right--;
        }
    }
    
    return maxArea;
}